### DISCUSSION TITLE
Zbus vs mailbox

### DISCUSSION THREAD

**Discussion by rhvarrier** [zephyrproject-rtos/zephyr#72444] [Q&A] [answered]
Hello, 

I want to use Zephyr to build a distributed system that needs to communicate between
- different SoC over a network (SPI, Serial or Ethernet)
- different threads running in the same SoC
- different cores of the same SoC

I'm looking for a abstraction that allows to do all of the above use a siliar API. 
Is Zbus suitable for this purpose or should I rather use message queue or mailbox with a combination of  IPM?
I didn't see any documentation comparing zbus to the existing message passing protocols implemented by Zephyr 
I also understand there is MBOX that seems to address the same problem. So what is the difference between these solutions?

**Comment by koffes** [comment, marked as answer]
As far as I know, there is currently no abstraction which allows all of these.

For the simplest "different threads running in the same SoC" there are as you may know numerous options:
Low level kernel services comparison:
https://docs.zephyrproject.org/latest/kernel/services/index.html#data-passing

Higher level Zbus:
https://docs.zephyrproject.org/latest/services/zbus/index.html

Note that there is an effort on Zbus to support different cores on the same SoC, so that should at least cater to two of your three requirements in the future.

As you said MBOX:
https://docs.zephyrproject.org/latest/hardware/peripherals/mbox.html#api-reference. This is the one I am least familiar with and it seems to be only for inter-processor comms.
